client/store/types/chapter.ts

```diff
@@ -0,0 +1,45 @@
+import { ThunkAction } from 'redux-thunk';
+import { AppStoreState } from '../reducers';
+
+export const FETCH_START = 'fcc/chapter/CHAPTER_START';
+export const FETCH_SUCCESS = 'fcc/chapter/CHAPTER_SUCCESS';
+export const FETCH_FAIL = 'fcc/chapter/CHAPTER_FAIL';
+
+export interface IChapterModal {
+  id?: string;
+  name: string;
+  description: string;
+  category: string;
+}
+
+export interface IChapterStoreState extends IChapterModal {
+  loading: boolean;
+  error: string; // Should reflect a generic Error Type here
+}
+
+interface IChapterFetchStartAction {
+  type: typeof FETCH_START;
+}
+
+interface IChapterFetchSuccessAction {
+  type: typeof FETCH_SUCCESS;
+  data: IChapterModal;
```

Comment:
Wouldn't it be better to use generic Flux names -> `type` and `payload`?

Reply:
Dindn't know that it's kind of a norm. Thanks for letting me know. Will do it 🎉